test/controllers/api/v2/provisioning_templates_controller_test.rb

```diff
@@ -200,5 +200,50 @@ class Api::V2::ProvisioningTemplatesControllerTest < ActionController::TestCase
       get :global_registration
       assert_response :internal_server_error
     end
+
+    test "should pass permitted params to template" do
+      params = {
+        organization_id: taxonomies(:organization1).id,
+        location_id: taxonomies(:location1).id,
+        hostgroup_id: hostgroups(:common).id,
+      }
+
+      get :global_registration, params: params, session: set_session_user
+      assert_response :success
+
+      vars = assigns(:global_registration_vars)
+      assert_equal taxonomies(:organization1), vars[:organization]
+      assert_equal taxonomies(:location1), vars[:location]
+      assert_equal hostgroups(:common), vars[:hostgroup]
+      assert_equal users(:admin), vars[:user]
+    end
+
+    test "should not pass unpermitted params to template" do
+      params = {
+        not_allowed: 'something_not_allowed',
+      }
+
+      get :global_registration, params: params, session: set_session_user
+      assert_response :success
+      assert_nil assigns(:global_registration_vars)[:not_allowed]
+    end
+
+    test "should allow to extend permitted params" do
+      Foreman::Plugin.any_instance.stubs(:allowed_registration_vars).returns([:activation_key])
+      get :global_registration, params: { activation_key: 'one-two-three-key' }, session: set_session_user
+      assert_response :success
+      assert_equal 'one-two-three-key', assigns(:global_registration_vars)[:activation_key]
+    end
+
+    test 'should fail when resources are not found' do
+      get :global_registration, params: { organization_id: 999 }, session: set_session_user
+      assert_response :not_found
+
+      get :global_registration, params: { location_id: 999 }, session: set_session_user
+      assert_response :not_found
+
+      get :global_registration, params: { hostgroup_id: 999 }, session: set_session_user
+      assert_response :unprocessable_entity
```

Comment:
Why is an incorrect hostgroup a HTTP 422 and location/org 404? I can't explain the difference since both call the same API's, right? From a user interaction perspective I think 422 is the more correct.

ID 999 can be used by some unlucky chance while 0 is generally not used since IDs start with 1. Would that be better?

Reply:
It's because of `set_taxonomy` in `app/controllers/concerns/application_shared.rb`.
For all other objects we render `422`, but for loc and org we render `404`

Reply:
that's probably because `/api/organization/0/$action` does not exist while `/api/$action?hostgroup_id=0` is wrong use of the existing endpoint

Reply:
Interesting. That suggests that the code to process the parameters is redundant and the current org/loc set by that logic is usable. Perhaps that would be even better. The code would have fewer conditionals. However, I won't block on that.